src/components/Calculator.js

```diff
@@ -1,27 +1,41 @@
-import React from 'react';
+import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
-const resultElement = <input type="number" className="result" value="0" disabled />;
+const keys = ['AC', '+/-', '%', '÷',
+  7, 8, 9, 'x',
+  4, 5, 6, '-',
+  1, 2, 3, '+',
+  0, '.', '='];
 
-class Calculator extends React.Component {
-  keys = ['AC', '+/-', '%', '÷',
-    7, 8, 9, 'x',
-    4, 5, 6, '-',
-    1, 2, 3, '+',
-    0, '.', '='];
+function CalculatorKey(props) {
+  const props1 = { ...props };
+  return (<input type="button" onClick={props1.clickHandler} className={(Number(props1.value) > 0) ? 'number' : 'key'} value={props1.value} />);
+}
+
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+    this.state = {};
+  }
 
-  keysElements = this.keys.map((key) => (<this.CalculatorKey value={key} key={key.toString()} />));
+  keysElements() {
+    return (keys.map((key) => (
+      <CalculatorKey value={key} key={key.toString()} clickHandler={this.clickHandler} />)));
+  }
 
-  // eslint-disable-next-line class-methods-use-this
-  CalculatorKey(props) {
-    const props1 = { ...props };
-    return (<input type="button" className={(Number(props1.value) > 0) ? 'number' : 'key'} value={props1.value} />);
+  clickHandler(e) {
+    this.setState((state) => calculate(state, e.target.value));
+    // this.calculatorData = calculate(e.target, e.target.value);
   }
 
   render() {
+    const keys = this.keysElements();
+    const { next, total } = this.state;
     return (
       <div className="calculatorContainer">
-        {resultElement}
-        {this.keysElements}
+        <input type="number" className="result" value={next || total || 0} disabled />
+        {keys}
       </div>
```

Comment:
- [ ] When I try to perform modulo or division operations involving 0, the calculator output goes blank, but React gives a warning in the console. Kindly note that the exception messages in the `operate.js` file are expected to be displayed to the user so they can know what went wrong, instead of having an empty screen. Kindly fix that.

  ![modulo-1](https://user-images.githubusercontent.com/54848363/182706987-85d619d5-1b80-4ff6-9be1-948743308d63.jpg)